lib/dialects/postgres/query/compiler.js

```diff
@@ -65,6 +70,40 @@ class QueryCompiler_PG extends QueryCompiler {
     return value ? ` returning ${this.formatter.columnize(value)}` : '';
   }
 
+  _ignore(columns) {
+    return ` on conflict (${this.formatter.columnize(columns)}) do nothing`;
+  }
+
+  _merge(updates, columns, insert) {
+    let sql = ` on conflict (${this.formatter.columnize(
+      columns
+    )}) do update set `;
+    if (updates) {
+      const updateData = this._prepUpdate(updates);
+      if (typeof updateData === 'string') {
+        sql += updateData;
+      } else {
+        sql += updateData.join(',');
+      }
+
+      return sql;
+    } else {
+      const insertData = this._prepInsert(insert);
+      if (typeof insertData === 'string') {
+        throw new Error(
+          'If using merge with a raw insert query, then updates must be provided'
+        );
+      }
+
+      sql += insertData.columns
+        .map((column) => this.formatter.wrapString(column.split('.').pop()))
```

Comment:
Why here needed `.split('.').pop()`?

Reply:
I was thinking that the column specified for the insert may be of the form `tableName.columnName`, in which case I only wanted the column name. I guess specifying the table name might not be supported for inserts anyway?